--- a/bus/bus.c
+++ b/bus/bus.c
@@ -147,7 +147,7 @@
 	adap = kzalloc(sizeof(*adap), GFP_KERNEL);
 	if (!adap)
 		return ERR_PTR(-ENOMEM);
-	adap->id = ida_simple_get(&gip_adapter_ida, 0, 0, GFP_KERNEL);
+	adap->id = ida_alloc(&gip_adapter_ida, GFP_KERNEL);
 	if (adap->id < 0) {
 		err = adap->id;
 		goto err_put_device;
@@ -178,7 +178,7 @@
 err_destroy_queue:
 	destroy_workqueue(adap->clients_wq);
 err_remove_ida:
-	ida_simple_remove(&gip_adapter_ida, adap->id);
+	ida_free(&gip_adapter_ida, adap->id);
 err_put_device:
 	put_device(&adap->dev);
 	return ERR_PTR(err);
@@ -214,7 +214,7 @@
 		device_unregister(&client->dev);
 	}
 
-	ida_simple_remove(&gip_adapter_ida, adap->id);
+	ida_free(&gip_adapter_ida, adap->id);
 	destroy_workqueue(adap->clients_wq);
 	dev_dbg(&adap->dev, "%s: unregistered\n", __func__);
 	device_unregister(&adap->dev);
